scaler/handlers.go

```diff
@@ -154,41 +144,44 @@ func (e *impl) GetMetricSpec(
 }
 
 func (e *impl) GetMetrics(
-	ctx context.Context,
+	_ context.Context,
 	metricRequest *externalscaler.GetMetricsRequest,
 ) (*externalscaler.GetMetricsResponse, error) {
-	lggr := e.lggr.WithName("GetMetrics")
 	sor := metricRequest.ScaledObjectRef
-
-	//namespacedName := k8s.NamespacedNameFromScaledObjectRef(sor)
-	//metricName := namespacedName.Name
-
-	scalerMetadata := sor.GetScalerMetadata()
-	metricName, labels, agg, err := util.GetMetricQuery(lggr, scalerMetadata, e.metricParser)
+	value, err := e.getMetric(sor)
 	if err != nil {
 		return nil, err
 	}
 
-	opOverTime := util.GetOperationOvertTime(lggr, scalerMetadata)
-	value, found, err := e.metricStore.Get(metricName, labels, opOverTime, agg)
-	lggr.Info("got metric value: ", "name", metricName, "labels", labels, "value", value, "found", found, "error", err)
-	if !found {
-		return nil, fmt.Errorf("not found")
-	}
-	if err != nil {
-		return nil, err
-	}
-	value = util.ClampValue(lggr, value, scalerMetadata)
-	e.metricsExporter.SetMetricValueClamped(string(metricName), fmt.Sprint(labels), string(opOverTime), string(agg), sor.GetName(), sor.GetNamespace(), value)
-
 	res := &externalscaler.GetMetricsResponse{
 		MetricValues: []*externalscaler.MetricValue{
 			{
 				MetricName:  metricRequest.GetMetricName(),
-				MetricValue: int64(math.Round(value)),
+				MetricValue: int64(math.Ceil(value)),
```

Comment:
side note: we should fix the external scaler interface, KEDA internal scalers support float here, eg:  https://github.com/kedacore/keda/blob/cd595a1c3fd56ca17806ff8331a1cd6ed7549e12/pkg/scalers/activemq_scaler.go#L208

Reply:
yep, I wanted to ask about that.. so I can make a pr to the proto as well and once it's there, throw away the float -> int function. I used the ceil instead of Round here to make the scale out use case faster

Reply:
If you ment kedacore proto, then yeah. But we should also maintain a backwards compatibility somehow, there are existing external scalers that rely on int param there.

Reply:
pr upstream - https://github.com/kedacore/keda/pull/6482